Stop resolving border's current color to some hardcoded value

obyknovenius opened the pull request:
#### 0aa032e26bbd7e95d38cc18aa16df600225f6ec7
<pre>
Stop resolving border&apos;s current color to some hardcoded value
<a href="https://bugs.webkit.org/show_bug.cgi?id=258725">https://bugs.webkit.org/show_bug.cgi?id=258725</a>

Reviewed by Tim Nguyen.

We currently return some hardcoded value when resolving
current color for border properties.
I can&apos;t think of any good reason for doing this.
The result can be unexpected for web developers.

* Source/WebCore/css/html.css:
(hr):
(#if defined(WTF_PLATFORM_IOS_FAMILY) &amp;&amp; WTF_PLATFORM_IOS_FAMILY):
* Source/WebCore/rendering/style/RenderStyle.cpp:
(WebCore::RenderStyle::colorResolvingCurrentColor const):

Canonical link: <a href="https://commits.webkit.org/266134@main">https://commits.webkit.org/266134@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bfa9c21a70baacd4621e9cc3fa839f4362bd7113

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/12988 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/13307 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/13638 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/14726 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/12389 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/13056 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/15812 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/13332 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/15084 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/13154 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/13849 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/10984 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/15188 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/11137 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/11739 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/18809 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/12212 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/11905 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/15122 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/12380 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/10267 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/11645 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3182 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/15962 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/12222 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Source/WebCore/rendering/style/RenderStyle.cpp:
```diff
@@ -2398,24 +2398,6 @@ Color RenderStyle::colorResolvingCurrentColor(CSSPropertyID colorProperty, bool
-        }();
-
-        if (!visitedLink && (borderStyle == BorderStyle::Inset || borderStyle == BorderStyle::Outset || borderStyle == BorderStyle::Ridge || borderStyle == BorderStyle::Groove))
-            return { SRGBA<uint8_t> { 238, 238, 238 } };
```
This color dates back to https://commits.webkit.org/4@main, through many many refactorings :)

nt1m reviewed (commented):
Looks fine, but can you add a WPT reftest? Chrome also inherited this bug from us so it's probably worth adding this to WPT:

```html
<style>
div {
    border-width: 5px;
    border-color: currentColor;
    color: green;
    margin-bottom: 2px;
}
</style>
<p>Test passes if all the border colors are green</p>
<div style="border-style: inset">
inset
</div>
<div style="border-style: outset">
outset
</div>
<div style="border-style: ridge">
ridge
</div>
<div style="border-style: groove">
groove
</div>
```

Reference:
```html
<style>
div {
    border-width: 5px;
    border-color: green;
    margin-bottom: 2px;
}
</style>
<p>Test passes if all the border colors are green</p>
<div style="border-style: inset">
inset
</div>
<div style="border-style: outset">
outset
</div>
<div style="border-style: ridge">
ridge
</div>
<div style="border-style: groove">
groove
</div>
```

nt1m reviewed (approved):
Looks great, thanks!

obyknovenius committed:
Stop resolving border's current color to some hardcoded value
https://bugs.webkit.org/show_bug.cgi?id=258725

Reviewed by Tim Nguyen.

We currently return some hardcoded value when resolving
current color for border properties.
I can't think of any good reason for doing this.
The result can be unexpected for web developers.

* Source/WebCore/css/html.css:
(hr):
(#if defined(WTF_PLATFORM_IOS_FAMILY) && WTF_PLATFORM_IOS_FAMILY):
* Source/WebCore/rendering/style/RenderStyle.cpp:
(WebCore::RenderStyle::colorResolvingCurrentColor const):

Canonical link: https://commits.webkit.org/266134@main
